lib/datasource/gradle-version/index.ts

```diff
@@ -1,63 +1,66 @@
-import { ExternalHostError } from '../../types/errors/external-host-error';
-import { Http } from '../../util/http';
-import { HttpError } from '../../util/http/types';
 import { regEx } from '../../util/regex';
 import * as gradleVersioning from '../../versioning/gradle';
+import { Datasource } from '../datasource';
 import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
 import type { GradleRelease } from './types';
 
-export const id = 'gradle-version';
-export const customRegistrySupport = true;
-export const defaultRegistryUrls = ['https://services.gradle.org/versions/all'];
-export const defaultVersioning = gradleVersioning.id;
-export const registryStrategy = 'merge';
+export class GradleVersionDatasource extends Datasource {
+  static readonly id = 'gradle-version';
 
-const http = new Http(id);
+  constructor() {
+    super(GradleVersionDatasource.id);
+  }
 
-const buildTimeRegex = regEx(
-  '^(\\d\\d\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\+\\d\\d\\d\\d)$'
-);
+  readonly defaultRegistryUrls = ['https://services.gradle.org/versions/all'];
 
-function formatBuildTime(timeStr: string): string | null {
-  if (!timeStr) {
-    return null;
-  }
-  if (buildTimeRegex.test(timeStr)) {
-    return timeStr.replace(buildTimeRegex, '$1-$2-$3T$4:$5:$6$7');
-  }
-  return null;
-}
+  readonly defaultVersioning = gradleVersioning.id;
+
+  readonly registryStrategy = 'merge';
+
+  private static readonly buildTimeRegex = regEx(
+    '^(\\d\\d\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\d\\d)(\\+\\d\\d\\d\\d)$'
+  );
 
-export async function getReleases({
-  registryUrl,
-}: GetReleasesConfig): Promise<ReleaseResult> {
-  let releases: Release[];
-  try {
-    const response = await http.getJson<GradleRelease[]>(registryUrl);
-    releases = response.body
-      .filter((release) => !release.snapshot && !release.nightly)
-      .map((release) => ({
-        version: release.version,
-        releaseTimestamp: formatBuildTime(release.buildTime),
-        ...(release.broken && { isDeprecated: release.broken }),
-      }));
-  } catch (err) {
-    if (
-      err instanceof HttpError &&
-      err.response?.url === defaultRegistryUrls[0]
-    ) {
-      throw new ExternalHostError(err);
+  async getReleases({
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult> {
+    let releases: Release[];
+    try {
+      const response = await this.http.getJson<GradleRelease[]>(registryUrl);
+      releases = response.body
+        .filter((release) => !release.snapshot && !release.nightly)
+        .map((release) => ({
+          version: release.version,
+          releaseTimestamp: GradleVersionDatasource.formatBuildTime(
+            release.buildTime
+          ),
+          ...(release.broken && { isDeprecated: release.broken }),
+        }));
+    } catch (err) {
+      this.handleGenericErrors(err);
     }
-    throw err;
```

Comment:
I didn't migrate this error handling code, because I think the default error handling is better. Thoughts?

https://github.com/renovatebot/renovate/blob/d4a22c7f3c874a7d451022e4680b66abca196796/lib/datasource/datasource.ts#L39-L51

Reply:
Seems ok to me.